CMakeLists.txt

```diff
@@ -303,6 +303,7 @@ endif ()
 
 
 if (MSVC)
+    add_definitions (-DNOMINMAX)
```

Comment:
I'm wary of this change. It will surely protect a build of OIIO itself, but I fear it will mask errors in our public headers that will bite users and developers who use our headers in other projects that don't provide this definition.

I think we should NOT define this project-wide, but rather hunt down and fix all of our individual header files where this needs to be set.